flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -197,13 +197,18 @@ BufferAndBacklog pollBuffer() {
 				return null;
 			}
 
+			boolean isExactlyOnceCheckpointBarrier = buffer.getDataType().isExactlyOnceCheckpointBarrier();
```

Comment:
use `buffer.getDataType() == Buffer.DataType.EXACTLY_ONCE_CHECKPOINT_BARRIER` instead.

Reply:
I add a util method in Buffer.DataType to judge if a Buffer is an ExactlyOnceCheckpointBarrier